library/src/main/java/com/parse/internal/signpost/basic/HttpURLConnectionProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.parse.internal.signpost.basic;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Singleton class to provide HttpURLConnection, using native HttpURLConnection or OkHttp
+ */
+public class HttpURLConnectionProvider {
```

Comment:
Let's rename this to `HttpURLConnectionClient` to match `com.apache.HttpClient`.

It also looks like you're using 4 space indentation instead of 2 space.